sqlglot/dialects/teradata.py

```diff
@@ -13,6 +13,29 @@
 from sqlglot.tokens import TokenType
 
 
+def _date_add_sql(
+    kind: t.Literal["+", "-"],
+) -> t.Callable[[Teradata.Generator, exp.DateAdd | exp.DateSub], str]:
+    def func(self: Teradata.Generator, expression: exp.DateAdd | exp.DateSub) -> str:
+        this = self.sql(expression, "this")
+        unit = expression.args.get("unit")
+
+        if isinstance(expression.expression, exp.Neg):
+            kind_to_op = {"+": "-", "-": "+"}
+            simplify_expr = expression.expression.this
+        else:
+            kind_to_op = {"+": "+", "-": "-"}
+            simplify_expr = expression.expression
+        expression = self._simplify_unless_literal(simplify_expr)
+        if not isinstance(expression, exp.Literal):
+            self.unsupported("Cannot add non literal")
+
+        expression.args["is_string"] = True
```

Comment:
```suggestion
        expression.args.set("is_string", True)
```

Reply:
Yeah that was a typo, it needs to be `expression.set("is_string", True)`

Reply:
that works, thanks!